pandas/io/formats/html.py

```diff
@@ -343,8 +345,21 @@ def _write_body(self, indent):
             else:
                 self._write_regular_rows(fmt_values, indent)
         else:
-            for i in range(min(len(self.frame), self.max_rows)):
-                row = [fmt_values[j][i] for j in range(len(self.columns))]
+            truncate_h = self.fmt.truncate_h
+            truncate_v = self.fmt.truncate_v
+            ncols = len(self.fmt.tr_frame.columns)
+            nrows = len(self.fmt.tr_frame)
+
+            row = []
```

Comment:
can you give some comments on what is happening here, would not object to a self.write_row?

Reply:
@jreback 
The added code is copy and pasted from `self._write_regular_rows()`

But for the `index=False` case, the (row) index value is not added to `row`.  The assignment to `dot_col_ix` and the value of `nindex_levels` in the `self.write_tr` call is also changed to account for this.

Currently the truncation tests in `tests\io\formats\test_to_html.py` are being skipped. Without test coverage the values of `dot_col_ix` and `nindex_levels` can take on any value and pass the tests.

To avoid potential regression on the notebook display codepath (`index=True`),  i have duplicated the code rather than make any changes to `self._write_regular_rows()` 

Tests could be added but I wanted to avoid going out of scope on this PR.

The existing code for the `index=False` case was not in a function unlike the code for the `index=True` cases. However it was only 3 lines.

So I would agree that a `self.write_row` function would now be a good idea if it was unlikely that a future refactoring was not to use `self._write_regular_rows()` instead.

How should I proceed?

Reply:
> Tests could be added but I wanted to avoid going out of scope on this PR.

can you do a pre-cursor PR which locks down the tests first?

Reply:
@simonjayhawkins i mean comments in the code